lib/resources.js

```diff
@@ -605,6 +603,86 @@ const buildElasticQueryForKeywords = function (params) {
   return query
 }
 
+const filterClausesForDateParams = function (filters) {
+  const hasDateAfter = typeof filters.dateAfter === 'number'
+  const hasDateBefore = typeof filters.dateBefore === 'number'
+
+  if (!hasDateAfter && !hasDateBefore) {
+    return []
+  } else if (hasDateAfter && hasDateBefore) {
+    /**
+     * When dateAfter and dateBefore used together, we want to match on:
+     *   dateStartYear != null && (
+     *     dateStartYear >= filters.dateAfter ||
+     *     dateEndYear >= filters.dateAfter
+     *   ) && (
+     *     dateStartYear <= filters.dateBefore ||
+     *     dateEndYear <= filters.dateBefore
+     *   )
+    */
+    return [{
+      clause: {
+        bool: {
+          must: [
+            { exists: { field: 'dateStartYear' } },
+            {
+              bool: {
+                should: [
+                  { range: { dateStartYear: { gte: filters.dateAfter } } },
+                  { range: { dateEndYear: { gte: filters.dateAfter } } }
+                ]
+              }
+            },
+            {
+              bool: {
+                should: [
+                  { range: { dateStartYear: { lte: filters.dateBefore } } },
+                  { range: { dateEndYear: { lte: filters.dateBefore } } }
+                ]
+              }
+            }
+          ]
+        }
+      }
+    }]
+  } else if (hasDateAfter) {
+    /**
+     * When dateAfter used alone, we want to match on:
+     *   dateStartYear != null && (
+     *     dateStartYear >= filters.dateAfter ||
+     *     dateEndYear >= filters.dateAfter
+     *   )
+    */
+    return [{
+      clause: {
+        bool: {
+          must: [
+            { exists: { field: 'dateStartYear' } },
+            {
+              bool: {
+                should: [
+                  { range: { dateStartYear: { gte: filters.dateAfter } } },
+                  { range: { dateEndYear: { gte: filters.dateAfter } } }
+                ]
+              }
+            }
+          ]
+        }
+      }
+    }]
+  } else if (hasDateBefore) {
+    /**
+     * When dateBefore used alone, we only need to match on:
+     *   dateStartYear <= filters.dateAfter
```

Comment:
I think this should say `filters.dateBefore`

Reply:
Fixed! Good catch